pallets/aleph/src/traits.rs

```diff
@@ -26,3 +26,17 @@ where
         })
     }
 }
+
+/// Provides the current total issuance.
+pub trait TotalIssuanceProvider<T: Config> {
+    fn get() -> Balance;
+}
+
+impl<T> TotalIssuanceProvider<T> for pallet_balances::Pallet<T>
```

Comment:
This could be moved to runtime/lib.rs and implemented for the new struct. The trait definition could be moved to the primitives

Reply:
done